synapse/lib/stormtypes.py

```diff
@@ -20,6 +21,8 @@
 
 logger = logging.getLogger(__name__)
 
+urlpass_re = regex.compile(r'(.+://.+:).+(@)')
```

Comment:
This behavior should be a convenience function inside of synapse.lib.urlhelp that chops the url, and replaces the passwd with a value (default to a string of asterisks) and returns the rejoined value.

Reply:
specifically, it should only do the passwd replacement if the passwd is present in the choped url